api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -117,6 +117,14 @@ message ExtAuthz {
   // When this field is true, Envoy will include the peer X.509 certificate, if available, in the
   // :ref:`certificate<envoy_api_field_service.auth.v3.AttributeContext.Peer.certificate>`.
   bool include_peer_certificate = 10;
+
+  // By default, the timeout for a check request starts counting when the request is becomes active.
+  // If this is set, the timeout will start when check request is created. There can be a
+  // difference between the two, if requests to the ext authz server become pending due to circuit
+  // breaking (i.e. by default, the duration while the request is pending is not counted towards
+  // the configured timeout). This is ignored for google_grpc service type, as it does not use the envoy
+  // io facilities.
```

Comment:
Another way of approaching this would be to change the existing behavior (under runtime flag guard) to start the timeout from when the request is created, arguably this is the behavior end users expect. We've had a recent discussion with some of the gRPC folks on some other timeouts that fit this pattern (@markdroth @dfawley).

In your proposal, you add an internal timer; if we use this exclusively, this would avoid having to have end users worry about the difference between Envoy and Google gRPC implementations.

Reply:
Thanks for the feedback @htuch !

i agree that using the runtime is preferable. the only reason i added it as a config option is for backwards compatibility.

are there any other stake holders in the community that you think might object? or should I go ahead and change this?

should the new behavior (internal timer) be the default, with runtime flag to revert? or the other way around?

Reply:
My vote would be runtime flag, default on, since most folks won't be relying on behavior that occurs during circuit breaker activation. That said, it is kind of hard to reason about the ideal behavior here under circuit breaker overload; currently we queue and drop when the queue is too long, we'll be encouraging more drop due to timeouts. @mattklein123 WDYT?

Reply:
Unless there are objections - I will adjust the code to perform the new behavior by default, with runtime flag to turn off.

Reply:
> My vote would be runtime flag, default on, since most folks won't be relying on behavior that occurs during circuit breaker activation. That said, it is kind of hard to reason about the ideal behavior here under circuit breaker overload; currently we queue and drop when the queue is too long, we'll be encouraging more drop due to timeouts. @mattklein123 WDYT?

Sorry. I very much missed this ping and I wished I hadn't. FWIW I would have probably figured out how to make this change in a more core way (potentially via a different optional param fed to the async client), mainly so that it would apply to other side calls also. Since I think ratelimit and other services would face the same problem. We can potentially consider doing this in a follow up.